src/Host.cpp

```diff
@@ -42,6 +42,140 @@
 #include <zip.h>
 #include "post_guard.h"
 
+stopWatch::stopWatch()
+: mIsInitialised(false)
+, mIsRunning(false)
+, mIsPersistent(false)
+, mEffectiveStartDateTime()
+, mElapsedTime()
+{
+    mEffectiveStartDateTime.setTimeSpec(Qt::UTC);
+}
+
+bool stopWatch::start()
+{
+    if (!mIsInitialised) {
+        mIsInitialised = true;
+        mEffectiveStartDateTime = QDateTime::currentDateTimeUtc();
+        mIsRunning = true;
+        return true;
+    }
+
+    if (mIsRunning) {
+        // Nothing to do, already running
+        return false;
+    }
+
+    // Is stopped, so subtract elapsed time from current and set that to be the
+    // effective start time:
+    mEffectiveStartDateTime = QDateTime::currentDateTimeUtc().addMSecs(-mElapsedTime);
+    mIsRunning = true;
+    return true;
+}
+
+bool stopWatch::stop()
+{
+    if (!mIsInitialised) {
+        // Nothing to do, never started
+        return false;
+    }
+
+    if (!mIsRunning) {
+        // Nothing to do, already stopped
+        return false;
+    }
+
+    // Is running - so stop and note time:
+    mElapsedTime = mEffectiveStartDateTime.msecsTo(QDateTime::currentDateTimeUtc());
+    mIsRunning = false;
```

Comment:
:beetle: Oh, forgot a `return true;` there - surprised that wasn't picked up - or perhaps "control reaches end of a non-void function" only triggers a warning...

Reply:
Fixed this (and a couple of other) details...